frontend/src/metabase/modes/types.ts

```diff
@@ -89,12 +62,39 @@ type AlwaysDefaultClickAction = Omit<
   defaultAlways: true;
 };
 
-// TODO [#26836]: unify this and frontend/src/metabase-types/types/Visualization.ts
 export type ClickAction = RegularClickAction | AlwaysDefaultClickAction;
 
-export type DrillOptions = {
-  question: Question;
-  clicked?: ClickObject;
+export type Drill = (options: ClickActionProps) => ClickAction[];
```

Comment:
nit: We could also change the click types to have an explicit `type` property and create a union of these types. This should eliminate the need for some of the type guards